build.sbt

```diff
@@ -43,90 +24,28 @@ ThisBuild / githubWorkflowAddedJobs ++= Seq(
       WorkflowStep.Sbt(List("coverage", "test", "coverageAggregate")),
       WorkflowStep.Run(List("bash <(curl -s https://codecov.io/bash)"))
     )
-  ),
-  WorkflowJob(
-    id = "publish-site",
-    name = "Publish site",
-    cond = {
-      val predicate = RefPredicate.StartsWith(Ref.Tag("v"))
-      val publicationCond = GenerativePlugin.compileBranchPredicate("github.ref", predicate)
-      Some(s"github.event_name != 'pull_request' && $publicationCond")
-    },
-    scalas = List("2.13.8"),
-    steps = WorkflowStep.Checkout ::
-      WorkflowStep.SetupJava(List(JavaSpec.temurin("11"))) ++
-      githubWorkflowGeneratedCacheSteps.value :+
-      WorkflowStep.Sbt(List("docs/makeSite")) :+
-      WorkflowStep.Use(
-        UseRef.Public("JamesIves", "github-pages-deploy-action", "3.7.1"),
-        params = Map(
-          "GITHUB_TOKEN" -> "${{ secrets.GITHUB_TOKEN }}",
-          "BRANCH" -> "gh-pages",
-          "FOLDER" -> "docs/target/site"
-        )
-      )
   )
 )
 
-ThisBuild / licenses := List(("MIT", url("http://opensource.org/licenses/MIT")))
+ThisBuild / licenses := List(License.MIT)
 
-lazy val jvmVersionSettings = VersionNumber(sys.props("java.version")) match {
-  case v if v.matchesSemVer(SemanticSelector(">1.8")) =>
-    Def.settings(
-      scalacOptions ++= {
-        val isScala211 = CrossVersion.partialVersion(scalaVersion.value).contains((2, 11))
-        if (isScala211) List("-target:jvm-1.8") else List("-release", "8")
-      },
-      // Suppresses problems with Scaladoc @throws links
-      Compile / doc / scalacOptions ++= Seq("-no-link-warnings"),
-      javacOptions ++= Seq("--release", "8")
-    )
-  case _ => Def.settings()
-}
+lazy val root = tlCrossRootProject.aggregate(core, bench)
 
-lazy val root = project
-  .in(file("."))
-  .aggregate(core.jvm, core.js, bench)
-  .enablePlugins(NoPublishPlugin)
-  .settings(scalaVersion := "2.13.8")
+lazy val docs =
+  project.in(file("site")).enablePlugins(TypelevelSitePlugin).dependsOn(core.jvm, bench)
```

Comment:
The `TypelevelSitePlugin` is documented here: https://typelevel.org/sbt-typelevel/site.html

To preview locally use `docs/tlSitePreview` in sbt.